### ISSUE TITLE
[Utopian Bug Hunting] Cannot Login to Localhost When Using Custom SC2 App

### ISSUE THREAD

**Issue by eastmaels** [utopian-io/utopian.io#305]
#### 1. Expected behavior

User should be able to login to localhost (https://localhost:3000) when using custom/own SC2 app.

#### 2. Actual behavior

User CANNOT login to localhost (https://localhost:3000) when using custom/own SC2 app.

It results to a **"Failed to create session"** error response after connecting to steemconnect.

#### 3. How to reproduce

Follow the steps indicated in the Utopian post [How To Run Utopian.io on Your Local Machine](https://steemit.com/utopian-io/@utopian-io/how-to-run-utopian-io-in-your-local-machine-here-is-how)

In the step **Create .env file**, the user/developer needs to generate keys for the ff. environment variables:
```
UTOPIAN_GITHUB_SECRET=GITHUB_SECRET_HERE
UTOPIAN_GITHUB_CLIENT_ID=GITHUB_CLIENT_ID_HERE
UTOPIAN_GITHUB_REDIRECT_URL=http://localhost:3000/github/callback
UTOPIAN_STEEMCONNECT_SECRET=STEEM_CONNECT_SECRET
MONGO_HOST=mongodb://localhost/utopian-io
```

It specifically indicates in the step:
> You can generate a Steem Connect secret by creating an application on https://v2.steemconnect.com/apps/create.
> Replace the Github values by creating a test app for your local environment.

In this author's case, I created an app named **familyprojects**. 

![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517282909/dd58zq67pam52rrgqses.png)

For the Github values, create your OAuth App via https://github.com/settings/developers.

![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517282006/t8ymgqmq4fxzthpohend.png)

After creating and generating your own keys for the above environment variables, run your local instances of the front-end (*utopian.io*) and the back-end (*api.utopian.io*).

##### 3.1 Back-end

![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517282033/ldulqnyekukddr6q6crz.png)
<sup>([source](https://steemit.com/utopian-io/@utopian-io/how-to-run-utopian-io-in-your-local-machine-here-is-how))</sup>

###### 3.1.1. Running Back-end (MongoDB)
![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517287569/ftorgqmsave4soh80fr4.png)

###### 3.1.2. Running Back-end (dev-server)
![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517287659/pxxabiqftrdl2ea9v2pn.png)

##### 3.2. Front-end

![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517282053/zlqp9xb46nekjabwdrp3.png)
<sup>([source](https://steemit.com/utopian-io/@utopian-io/how-to-run-utopian-io-in-your-local-machine-here-is-how))</sup>

###### 3.2.1. Running front-end
![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517287966/hfdpeaez7uoshsez3suy.png)

##### 4. Replication Steps

| Step | Screenshot |
| -- | -- |
| 1) Go to https://localhost:3000/ | |
| 2) Click on **Log in** | ![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517311346/uiudbnn7uaf3oulw5gd9.png) |
| 3) Click **Submit** to connect to SteemConnect | ![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517311388/ocwxhtiu3cxawiorxig6.png) |
| 4) Click **Continue** to authorize app to user your posting role | ![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517311442/qh60oldjxmvv6mxw9p4q.png) |
| 5) Input your Steemit username and **Active Key** | ![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517311527/e58qwpdlesoudvyfnymg.png) |
| 6) Click **SIGN IN** | |
| 7) Wait for Steemconnect to return a response | |

#### 5. Result

Browser displays **"Failed to create session"** and user is still not logged-in even if you refresh your page to https://localhost:3000.

![image.png](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517311573/abmhxfzgvj2vpvot4w8k.png)

* Browser: Chrome
* Operating system: Windows 10 Enterprise

#### 6. Recording Of The Bug

![cannot-login-to-local-env-using-own-sc2-app.gif](https://res.cloudinary.com/hpiynhbhq/image/upload/v1517288926/ueqkg40wqgvy4jmdplch.gif)

<br /><hr/><em>Posted on <a href="https://utopian.io/utopian-io/@eastmael/utopian-bug-hunting-cannot-login-to-local-environment-when-using-custom-sc2-app">Utopian.io -  Rewarding Open Source Contributors</a></em><hr/>

**Comment by eastmaels** [comment]
Fixed by PR #281.

**Comment by eastmaels** [comment]
And the instruction from the README.md

### Set SC Utopian App

Add the enviroment variable UTOPIAN_APP using the app name of your Steem Connect oauth application. You can create a SC app here https://v2.steemconnect.com/apps/create. The same app will have to be used and setup in the backend https://github.com/utopian-io/api.utopian.io.
